cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -110,6 +122,18 @@ void DrawableSubmap::Update(
     // for this submap.
     UpdateTransform();
   }
+  visibility_->setName(QStringLiteral("%1-%2.%3")
+                           .arg(trajectory_id_)
+                           .arg(submap_index_)
+                           .arg(metadata_version_));
+  visibility_->setDescription(
```

Comment:
You are trying to add a UI concept that rViz (or Qt) does not support natively. It will always be surprising to the user and hard to discover. And it does not add a ton of benefit here. I strongly suggest leaving it out.